src/libsync/propagatedownloadzsync.cpp

```diff
@@ -124,62 +117,56 @@ bool GETFileZsyncJob::finished()
     return true; // discard
 }
 
-void GETFileZsyncJob::seedFinished(void* zs) {
-    _zs = (struct zsync_state *)zs;
+void GETFileZsyncJob::seedFinished(void* zs)
+{
+    zsync_unique_ptr<struct zsync_state> _zs((struct zsync_state*)zs, [](struct zsync_state* zs) {
+        zsync_complete(zs);
```

Comment:
Is it safe to always call `zsync_complete`, even if `begin_receive` wasn't called yet?

Reply:
@ckamm You're not looking at latest push?

Reply:
Yes, the call to `zsync_end_receive` is what isn't.